x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/OrderBy.java

```diff
@@ -109,4 +110,9 @@ public void postAnalysisVerification(Failures failures) {
             }
         });
     }
+
+    @Override
+    public void postOptimizationVerification(Failures failures) {
+        failures.add(fail(this, "Unbounded sort not supported yet, please add a limit"));
```

Comment:
nit: maybe we could include the source in the error message to make this easier for users to understand.